### ISSUE TITLE
unused code or bug?

### ISSUE THREAD

**Issue by CanCanZeng** [GhiXu/ACMP#11]
Hi, I find a confusing code here, is this an unused code or a bug?
https://github.com/GhiXu/ACMP/blob/604e2347f8a7d302ebaa4cb619b0bf2e87527853/ACMP.cu#L588

this situation seems never happens. But if I change the code to `while (depth_perturbed < params.depth_min || depth_perturbed > params.depth_max);` the program will get stuck.

**Comment by CaptureFish** [comment]
Hi, I think this situation dose happen.
`float depth_perturbed = *depth;`
`const float depth_min_perturbed = (1 - perturbation) * depth_perturbed;`
`const float depth_max_perturbed = (1 + perturbation) * depth_perturbed;`
`depth_perturbed = curand_uniform(rand_state) * (depth_max_perturbed - depth_min_perturbed) + depth_min_perturbed;`
therefore it is possible that `depth_min_perturbed < params.depth_min` when `(*depth)=params.depth_min`.

**Comment by CanCanZeng** [comment]
@CaptureFish I agree that one of the two situations (`depth_perturbed < params.depth_min` or `depth_perturbed > params.depth_max`) may happen, but cannot **both** happen at the same time, so the `while `condition is useless.

**Comment by CaptureFish** [comment]
@CanCanZeng Hi, I am sorry for misunderstanding.  I think the reason for the stuck is that the random seed, namely the `rand_state`, keeps indentical in the do-while loop. Therefore the wrong `depth_perturbed `values keep unchanged.

**Comment by wenbo-xu-official** [comment]
I guess this code is used for debugging. Consider this situation when you use the params depth_max=-50 and depth_min=-10.
